CRM/Event/Form/Participant.php

```diff
@@ -413,19 +413,44 @@ public function preProcess() {
    */
   public function setDefaultValues(): array {
     if ($this->isOverloadFeesMode()) {
-      return CRM_Event_Form_EventFees::setDefaultValues($this);
+      if ($this->getEventID()) {
+        //get receipt text and financial type
+        $returnProperities = ['confirm_email_text', 'financial_type_id', 'campaign_id', 'start_date'];
+        CRM_Core_DAO::commonRetrieveAll('CRM_Event_DAO_Event', 'id', $this->getEventID(), $details, $returnProperities);
+        if (!empty($details[$this->getEventID()]['financial_type_id'])) {
+          $defaults['financial_type_id'] = $details[$this->getEventID()]['financial_type_id'];
+        }
+        if (!empty($details[$this->getEventID()]['confirm_email_text'])) {
+          $defaults['receipt_text'] = $details[$this->getEventID()]['confirm_email_text'];
+        }
+        if (!$this->getParticipantID()) {
+          $defaults['send_receipt'] = (strtotime(CRM_Utils_Array::value('start_date', $details[$this->getEventID()])) >= time()) ? 1 : 0;
+          $defaults['receive_date'] = date('Y-m-d H:i:s');
+        }
+      }
+
+      //CRM-11601 we should keep the record contribution
+      //true by default while adding participant
+      if ($this->getAction() === CRM_Core_Action::ADD && !$this->_mode && $this->_isPaidEvent) {
+        $defaults['record_contribution'] = 1;
+      }
+
+      //CRM-13420
+      if (empty($defaults['payment_instrument_id'])) {
```

Comment:
Feels a bit weird to do this in edit mode, probably no harm, but would be clearer to only set it if we're adding.

Reply:
Or maybe I'm misunderstanding how this works: When is isOverloadFeesMode true?

Reply:
@larssandergreen so overload mode is when the Participant_Form is being used to render the payment block as ajax. My hope is to move that to it's own class - & load that by ajax. I decided to separate out the event functionality first - here https://github.com/civicrm/civicrm-core/pull/27660 - but I was going to try to separate out the ajax load form after that.

The ajax mode form gets all it's values by url & has different expecations to the main form - so handling both together makes the whole thing a bit fraught - but once I separate them a lot will be cleaner

You can also see in the tpl is is basically one big `{if}{else}{/if}` - that is the overload mode rendering